### ISSUE TITLE
Feature request: findIdsAndCountAll method

### ISSUE THREAD

**Issue by nicoabie** [sequelize/sequelize#16003]
## Issue Creation Checklist

- [X] I understand that my issue will be automatically closed if I don't fill in the requested information
- [X] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

### Describe the feature you'd like to see implemented

There are cases where the Order is complex to compute as well as the joins to get the relevant data.
In these cases SQL engines will compute the outer query for getting the relevant data and then the inner queries for computing the order.
If we apply a limit to this, many of the rows will be discarded therefore a lot of the work done in joining the relevant data was useless

if we were to have a findIdsAndCountAll method we could first get the IDs of the rows we want to fetch and then with those IDs call find to compute the joins.

Basically findIdsAndCountAll will create a query that does filtering and joins only necessary for computer order and count BUT no joins to get actual data requested.

### Describe why you would like this feature to be added to Sequelize

It is part of what sequelize aims to solve.

### Is this feature dialect-specific?

- [X] No. This feature is relevant to Sequelize as a whole.
- [ ] Yes. This feature only applies to the following dialect(s): 

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [X] Yes, I have the time but I will need guidance.
- [ ] No, I don't have the time, but my company or I are [supporting Sequelize through donations on OpenCollective](https://opencollective.com/sequelize).
- [ ] No, I don't have the time, and I understand that I will need to wait until someone from the community or maintainers is interested in implementing my feature.

---

*Indicate your interest in the addition of this feature by adding the 👍 reaction. Comments such as "+1" will be removed.*

**Comment by ephys** [comment]
I'm struggling to understand the feature request. Could you provide an example of what the API would look like, as well as what the SQL it generates would be?

**Comment by nicoabie** [comment]
We are having a problem where sometimes we need to use "subQuery: false"

We are joining to many entities, filtering, sorting and limiting the result.

In this case it is very costly for the engine because we are joining to stuff that after the limit gets discarded.

So the idea behind "findIdsAndCountAll" was that it would have exactly the same API as "findAndCountAll" but I'll just return the IDs of the main entity, not joining on data that would be discarded after the limit.

So after calling it we could call just "find" with the IDs and join information and it would be faster.

Does it make sense?

So basically we are breaking the problem in two, filtering sorting and limiting in one query
getting all the fields with joins in the second query.

Again this is mostly useful if for some reason users need to "subQuery: false"

**Comment by ephys** [comment]
At the moment I think we need to focus on [rewriting findAll](https://github.com/sequelize/sequelize/issues/15260) before introducing new features that rely on findAll like this one
